workbench/src/sidebar.c

```diff
@@ -0,0 +1,1156 @@
+	if (gtk_tree_selection_get_selected(treesel, &model, &current))
+	{
+		/* Search through the parents upwards until we find the project node.
+		   Save everthing that's interesting in callers variables... */
+		do
+		{
+			gtk_tree_model_get(model, &current, FILEVIEW_COLUMN_DATA_ID, &dataid, -1);
+			gtk_tree_model_get(model, &current, FILEVIEW_COLUMN_ASSIGNED_DATA_POINTER, &data, -1);
+			if (data != NULL)
+			{
+				switch (dataid)
+				{
+					case DATA_ID_WB_BOOKMARK:
+						context->wb_bookmark = data;
+					break;
+					case DATA_ID_PROJECT:
+						context->project = data;
+					break;
+					case DATA_ID_PRJ_BOOKMARK:
+						context->prj_bookmark = data;
+					break;
+					case DATA_ID_DIRECTORY:
+						context->directory = data;
+					break;
+					case DATA_ID_NO_DIRS:
+						/* Has not got any data. */
+					break;
+					case DATA_ID_FOLDER:
+						context->folder = data;
+					break;
+					case DATA_ID_FILE:
+						context->file = data;
+					break;
+				}
+			}
+
+			has_parent = gtk_tree_model_iter_parent(model,
+							&parent, &current);
+			current = parent;
+		}while (has_parent);
+
+		return TRUE;
+	}
+	return FALSE;
+}
+
+
+/** Setup the sidebar.
+ *
+ **/
+void sidebar_init(void)
+{
+	GtkWidget *scrollwin;
+	GtkCellRenderer *renderer;
+	GtkTreeViewColumn *column;
+	GtkTreeSelection *sel;
+	PangoFontDescription *pfd;
+	GList *focus_chain = NULL;
+
+	sidebar.file_view_vbox = gtk_vbox_new(FALSE, 0);
+
+	/**** label ****/
+	sidebar.file_view_label = gtk_label_new (_("No workbench opened."));
+	gtk_box_pack_start(GTK_BOX(sidebar.file_view_vbox), sidebar.file_view_label, FALSE, FALSE, 0);
+
+	/**** tree view ****/
+
+	sidebar.file_view = gtk_tree_view_new();
+	g_signal_connect(sidebar.file_view, "row-activated", (GCallback)sidebar_filew_view_on_row_activated, NULL);
+
+	sidebar.file_store = gtk_tree_store_new(FILEVIEW_N_COLUMNS, G_TYPE_ICON, G_TYPE_STRING, G_TYPE_UINT, G_TYPE_POINTER);
+	gtk_tree_view_set_model(GTK_TREE_VIEW(sidebar.file_view), GTK_TREE_MODEL(sidebar.file_store));
+
+	renderer = gtk_cell_renderer_pixbuf_new();
+	column = gtk_tree_view_column_new();
+	gtk_tree_view_column_pack_start(column, renderer, FALSE);
+	gtk_tree_view_column_add_attribute(column, renderer, "gicon", FILEVIEW_COLUMN_ICON);
+
+	renderer = gtk_cell_renderer_text_new();
+	gtk_tree_view_column_pack_start(column, renderer, TRUE);
+	gtk_tree_view_column_add_attribute(column, renderer, "markup", FILEVIEW_COLUMN_NAME);
+
+	gtk_tree_view_append_column(GTK_TREE_VIEW(sidebar.file_view), column);
+
+	gtk_tree_view_set_headers_visible(GTK_TREE_VIEW(sidebar.file_view), FALSE);
+	gtk_tree_view_set_enable_search(GTK_TREE_VIEW(sidebar.file_view), TRUE);
+	gtk_tree_view_set_search_column(GTK_TREE_VIEW(sidebar.file_view), FILEVIEW_COLUMN_NAME);
+
+	pfd = pango_font_description_from_string(wb_globals.geany_plugin->geany_data->interface_prefs->tagbar_font);
+	gtk_widget_modify_font(sidebar.file_view, pfd);
+	pango_font_description_free(pfd);
```

Comment:
you could use `ui_widget_modify_font_from_string()` from Geany's utility function (which also would avoid the [GTK3 deprecation warning](https://travis-ci.org/geany/geany-plugins/jobs/266322308#L4954))